CommMicro.py

```diff
@@ -65,86 +65,113 @@ def getPath(fileName):
-        TopPlot = MplCanvas(self, width = 20, height =40, dpi=100)
-        BotPlot = MplCanvas(self, width = 20, height =40, dpi=100)
+        TopPlot = MplCanvas(self, width=20, height=40, dpi=100)
+        BotPlot = MplCanvas(self, width=20, height=40, dpi=100)
         self.xfDisp.ui.PlotTop.addWidget(TopPlot)
         self.xfDisp.ui.PlotBot.addWidget(BotPlot)
-        
+
         # call function setGOVal when strtBut is pressed
-        self.ui.StrtBut.clicked.connect(partial(self.setGOVal,TopPlot,BotPlot))  
+        self.ui.StrtBut.clicked.connect(partial(self.setGOVal, TopPlot, BotPlot))
 
         # call function getOldData when OldDatBut is pressed
-        self.ui.OldDatBut.clicked.connect(partial(self.getOldData,TopPlot,BotPlot))  
+        self.ui.OldDatBut.clicked.connect(partial(self.getOldData, TopPlot, BotPlot))
 
         # This doesn't work yet.
         # call function plotWindow when printPlot is pressed
-#        self.xfDisp.ui.printPlot.clicked.connect(self.plotWindow)
+        #        self.xfDisp.ui.printPlot.clicked.connect(self.plotWindow)
 
         # get CM IDs from FFt_math
-        self.CM_IDs=FFt_math.CM_IDs()
-
-        # set spinbox to 10 to start
-        self.ui.TimSpinBox.setValue(10)
+        self.CM_IDs = FFt_math.CM_IDs()
 
         # fill combo boxes
         for cmid in self.CM_IDs:
-          self.ui.CMComboBox.addItem(cmid)
+            self.ui.CMComboBox.addItem(cmid)
         self.ui.CavComboBox.addItem('Cavities 1-4')
         self.ui.CavComboBox.addItem('Cavities 5-8')
 
-        #start out with cavities 1-4 selected
-        self.checkboxes=[self.ui.cb1, self.ui.cb2, self.ui.cb3, self.ui.cb4]
-        for idx,cb in enumerate(self.checkboxes):
-          cb.setText(str(idx+1))
+        # start out with cavities 1-4 selected
+        self.checkboxes = [self.ui.cb1, self.ui.cb2, self.ui.cb3, self.ui.cb4]
+        for idx, cb in enumerate(self.checkboxes):
+            cb.setText(str(idx + 1))
 
         # check the first box so there's Something
         self.ui.cb1.setChecked(True)
         # call function if cavity select combo box changes
         self.ui.CavComboBox.activated.connect(self.ChangeCav)
 
+        # initialize checkbox counter
+        self.counter = 1
+
+        self.ui.comboBox_decimation.currentIndexChanged.connect(self.update_daq_setting)
+        self.ui.spinBox_buffers.valueChanged.connect(self.update_daq_setting)
+        self.update_daq_setting()
+
+        self.ui.cb1.stateChanged.connect(self.update_counter)
+        self.ui.cb2.stateChanged.connect(self.update_counter)
+        self.ui.cb3.stateChanged.connect(self.update_counter)
+        self.ui.cb4.stateChanged.connect(self.update_counter)
+
+    def update_daq_setting(self):
+
+        number_of_buffers = int(self.ui.spinBox_buffers.value())
+        decimation = int(self.ui.comboBox_decimation.currentText())
+        sampling_rate = DEFAULT_SAMPLING_RATE / decimation

```

Comment:
Does this need to be an int? because this is floating point division

Reply:
aaah, now I understand. You're right of course. 
Is it sufficient that the input for the decimation is limited to 1, 2 and 4 via the dropdown in the GUI? Or do we need to do something here?